src/hooks/useCart/useCart.tsx

```diff
@@ -206,7 +207,7 @@ export const useCart = (
             )
           );
         } else {
-          setError(e);
+          setError(new Error(e.message));
```

Comment:
This will discard the type of error, what's the reason for this change?

Reply:
reverting to use `setError(e)`, will be thrown to error boundary!